### PR TITLE
Updates to interpreter and executor to support new executor

### PR DESCRIPTION
**Pull request by angushtlam**
\#136
#135 
# Description

This PR updates the executor and interpreter to support the new executor rewrite.

# Change Log
Describe all changes made 

-`<FILE CHANGED>` : <Change> 
`src/interpreter/executor/interpret.js` - moved enums and the objects into respective files
`src/interpreter/executor/abstracts.js`- new file to reduce file size of executor
`src/interpreter/executor/enums.js` - new file to reduce file size of executor
`src/view/index.js` - updated the way console and variables get added to the state
`src/interpreter/executor/state.js` - updated the info that the state receives from the interpreter to save into the executor state
`src/view/console/index.js` - fixed console throwing a react error

# Manual Test Steps
Describe how a reviewer should test the changes. Include relevant workflows, API calls, etc...
**NOTE: All visual changes PR should include a screenshot**
The visualizer and console now works as intended

### CODE REVIEW

File: src/interpreter/executor/interpret.js [hunk 1]
```diff
@@ -302,7 +247,7 @@ function assigningVar(buffer, currentState, currExpression, inputLine) {
 }
 
 // If we are taking in console input
-function acceptingConsole(buffer, currExpression, inputLine) {
+function acceptingConsole(buffer, currentState, currExpression, inputLine) {
```
Comment (zsmoore) [inline src/interpreter/executor/interpret.js]: I don't think this is necessary to add.
If we have a console log we will always consume it and only call this function once so we do not care about input state.

We can delete the line `currentState = stateEnum.CONSUMED` and just return consumed always in the return tuple

File: src/interpreter/executor/interpret.js [hunk 2]
```diff
@@ -262,7 +207,7 @@ function evalState(buffer, currentState, currExpression, inputLine) {
             break;
 
         case stateEnum.ACCEPTING_CONSOLE:
-            resTup = acceptingConsole(buffer, currExpression, inputLine);
+            resTup = acceptingConsole(buffer, currentState, currExpression, inputLine);
```
Comment (zsmoore) [inline src/interpreter/executor/interpret.js]: See other comment.

**Review by zsmoore — CHANGES_REQUESTED** [review]

**Comment by zsmoore** [comment]
@angushtlam Do we want to update error output in this PR or handle it separately?

**Review by zsmoore — APPROVED** [review]
Pair programmed this with Angus.  Works perfectly. Great job!

**Comment by AlexPerri** [comment]
Are the Travis checks failing a problem right now or is it just not updated for the new changes Zach made in the interpreter?

**Review by AlexPerri — APPROVED** [review]
👍

**Comment by noahpoz** [comment]
I noticed that if there was a syntax error in the code, it seems to get locked in a execution state or something and I can't run any more code unless I refresh the page.

**Comment by zsmoore** [comment]
@noahpoz I think this has to do with error code not actually getting output yet

**Pull request merged**